### PR TITLE
feat: Updated release process to use goreleaser

### PR DESCRIPTION
**Pull request by cagiti**
This contributes towards #4120

### CODE REVIEW

File: jenkins-x.yml [hunk 1]
```diff
@@ -102,6 +102,16 @@ pipelineConfig:
...
+              - name: tag-release
+                image: jenkinsxio/jx:2.0.128
+                command: git
+                args: ['tag', '-a', '${inputs.params.version}', '-m', '${inputs.params.version}']
+
+              - name: push-tag
+                image: jenkinsxio/jx:2.0.128
+                command: git
+                args: ['push', 'origin', '${inputs.params.version}']
+
```
Comment (rawlingsj) [inline jenkins-x.yml]: We shouldn't need to do the git tag here as this already happens in the pipelinerunner

File: Makefile [hunk 2]
```diff
@@ -245,21 +245,15 @@ darwin: ## Build for OSX
 	chmod +x build/darwin/$(NAME)
 
 .PHONY: release
-release: clean build test-slow-integration linux darwin win arm ## Release the binary
```
Comment (pmuir) [inline Makefile]: I would probably delete the platform specific targets now - the `Makefile` is pretty busy already and any simplification would be great

**Pull request merged**